File: backend/app/schemas/ticket.py
from pydantic import BaseModel
from typing import Optional
from datetime import date


class TicketCreate(BaseModel):
    ticket_number: str
    pnr: Optional[str] = None
    airline_id: int
    booking_class: str
    travel_date: date
    issue_date: date
    passenger_name: Optional[str] = None
    origin_code: str
    destination_code: str
    base_fare: float
    taxes: float = 0
    total_fare: float
    currency: str = "USD"


class TicketRead(BaseModel):
    id: int
    ticket_number: str
    pnr: Optional[str]
    airline_id: int
    booking_class: str
    travel_date: date
    origin_code: str
    destination_code: str
    base_fare: float
    total_fare: float
    currency: str
    matched_deal_id: Optional[int]
    is_manually_matched: bool

    model_config = {"from_attributes": True}

File: frontend/src/types/ticket.ts
export interface Ticket {
  id: number;
  ticket_number: string;
  pnr: string | null;
  airline_id: number;
  booking_class: string;
  travel_date: string;
  origin_code: string;
  destination_code: string;
  base_fare: number;
  total_fare: number;
  currency: string;
  matched_deal_id: number | null;
  is_manually_matched: boolean;
}
